packages/react/src/themes/types.ts

```diff
@@ -131,15 +134,71 @@ export type ComponentAreaName =
 /**
  * A type for the generic color scheme of a component based on CSS property names
  */
-export type ColorScheme = Extendable<Record<ComponentAreaName, string>, string>
+export type ColorScheme<T extends string | number | symbol = ComponentAreaName> = Extendable<
+  Record<T, string>,
+  string
+>
+
+export type ColorSchemeMapping<
+  Scheme = ColorScheme,
+  Colors extends string | number | symbol = ColorNames
+> = ColorValues<Extendable<Scheme, string>, Colors> & {
+  default?: Extendable<Scheme, string>
+}
+
+export type StrictColorScheme<T extends string | number | symbol = ComponentAreaName> = Record<
+  T,
+  string
+>
 
-export type ColorSchemeMapping = ColorValues<Extendable<ColorScheme, string>> & {
-  default?: Extendable<ColorScheme, string>
+export type StrictColorSchemeMapping<
+  Scheme = StrictColorScheme,
+  Colors extends string | number | symbol = ColorNames
+> = ColorValues<Scheme, Colors> & {
+  default?: Scheme
 }
-export type ColorSchemeMappingOverrides = ColorValues<Partial<Extendable<ColorScheme, string>>> & {
+
+export type ColorSchemeMappingOverrides<
+  Scheme = ColorScheme,
+  Colors extends string | number | symbol = ColorNames
+> = ColorValues<Partial<Extendable<Scheme, string>>, Colors> & {
   default?: Partial<Extendable<ColorScheme, string>>
 }
 
+/**
+ * Teams theme specific interfaces
+ */
+export type TeamsContextualColors = {
```

Comment:
lets move them to separate `teams-types.ts` module

Reply:
Moved, thanks for the suggestion!